bpjs-debugger/src/main/java/il/ac/bgu/se/bp/execution/BPJsDebuggerImpl.java

```diff
@@ -277,12 +277,14 @@ public BooleanResponse nextSync() {
     }
 
     private boolean runNextSync() {
+        logger.info("runNextSync state: {0}", state.getDebuggerState());
+
+        if(!isThereAnyPossibleEvents()){
+            nextSyncOnNoPossibleEvents();
+        }
```

Comment:
no return?

Reply:
No, inside 'nextSyncOnNoPossibleEvents' we are waiting for some external event to be inserted, and then he will be inside the 'selectableEvents' set we will get in line 287.